infrastructure/core/main.tf

```diff
@@ -0,0 +1,79 @@
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+resource "azurerm_resource_group" "core" {
+  name     = "${var.prefix}-${var.environment}-rg"
+  location = var.location
+  tags     = var.tags
+}
+
+resource "azurerm_virtual_network" "core" {
+  name                = "${var.prefix}-${var.environment}-vnet"
+  resource_group_name = azurerm_resource_group.core.name
+  location            = azurerm_resource_group.core.location
+  address_space       = ["10.0.0.0/16"]
+  tags                = var.tags
+}
+
+resource "azurerm_subnet" "core" {
+  name                 = "${var.prefix}-${var.environment}-subnet-core"
+  resource_group_name  = azurerm_resource_group.core.name
+  virtual_network_name = azurerm_virtual_network.core.name
+  address_prefixes     = ["10.0.2.0/24"]
+  service_endpoints    = ["Microsoft.Sql", "Microsoft.Storage"]
+}
+
+resource "azurerm_storage_account" "core" {
+  name                     = "${var.prefix}${var.environment}strg"
+  resource_group_name      = azurerm_resource_group.core.name
+  location                 = azurerm_resource_group.core.location
+  account_tier             = "Standard"
+  account_replication_type = "GRS"
+  tags                     = var.tags
+
+  network_rules {
+    default_action             = "Deny"
+    ip_rules                   = ["100.0.0.1"]
+    virtual_network_subnet_ids = [azurerm_subnet.core.id]
+  }
+}
+
+resource "azurerm_key_vault" "core" {
+  name                        = "${var.prefix}-${var.environment}-kv"
+  location                    = azurerm_resource_group.core.location
+  resource_group_name         = azurerm_resource_group.core.name
+  enabled_for_disk_encryption = true
+  tenant_id                   = data.azurerm_client_config.current.tenant_id
+  soft_delete_retention_days  = 7
+  purge_protection_enabled    = false
+  sku_name                    = "standard"
+  tags                        = var.tags
+
```

Comment:
general question: when do we need

```hcl
  public_network_access_enabled   = false
```

Reply:
Good question - likely when I've locked down the other resources as part of the terraform story. We'll enable purge protection at some point as well but want to pick my battles :D